include/kernel.h

```diff
@@ -2013,6 +2014,127 @@ __syscall int k_futex_wake(struct k_futex *futex, bool wake_all);
 /** @} */
 #endif
 
+/**
+ * @defgroup event_apis Event APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/**
+ * Event Structure
+ * @ingroup event_apis
+ */
+
+struct k_event {
+	_wait_q_t         wait_q;
+	uint32_t          events;
+	struct k_spinlock lock;
+};
+
+#define K_EVENT_ACCUMULATE     0x00
+#define K_EVENT_OVERWRITE      0x01
+#define K_EVENT_DELIVERY_MASK  0x01
+
+#define K_EVENT_WAIT_ALL       0x00
+#define K_EVENT_WAIT_ANY       0x01
+#define K_EVENT_WAIT_MASK      0x01
+
+#define K_EVENT_PRE_CLEAR      0x02
+#define K_EVENT_POST_CLEAR     0x04
+
+#define Z_EVENT_INITIALIZER(obj) \
+	{ \
+	.wait_q = Z_WAIT_Q_INIT(&obj.wait_q), \
+	.events = 0 \
+	}
+/**
+ * @brief Initialize an event object
+ *
+ * This routine initializes an event object, prior to its first use.
+ *
+ * @param event Address of the event object.
+ *
+ * @return N/A
+ */
+
+__syscall void k_event_init(struct k_event *event);
```

Comment:
You declared it as syscall, but it is missing the verification function. Actually you need to make `k_event`  a k_object. Look https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/gen_kobject_list.py#L92

Reply:
Yes... this needs to be added to the list in `gen_kobject_list.py`.

Reply:
On it.